File: services/project/internal/data/models/comment.go
package models

import "time"

type TaskComment struct {
	ID        string    `db:"id"`
	Body      string    `db:"body"`
	TaskID    string    `db:"task_id"`
	UserID    string    `db:"user_id"`
	CreatedAt time.Time `db:"created_at"`
}

File: services/frontend/src/lib/task/comment.ts
import { User } from "../user/user";
import { Pagination } from "../utils/list";

export type TaskComment = {
  id: string;
  user: User;
  task_id: string;
  body: string;
  created_at: {
    seconds: number;
  };
};

export type TaskCommentCreation = Omit<
  TaskComment,
  "id" | "created_at" | "user"
> & {
  user_id: string;
};

export type TaskCommentFilter = Pagination &
  Pick<TaskComment, "task_id"> & { user_id: string };
